RIOT-OS/RIOT
# sx127x: add several NETOPT for GNRC LoRaWAN

### Contribution description
This PR add the following NETOPT to the SX127x driver:
- NETOPT_RANDOM: Get a random number from the LoRa device
- NETOPT_SYNCWORD: Get/set the LoRa PHY Syncword.
- NETOPT_RX_SYMBOL_TIMEOUT: Set the number of symbols the LoRa transceiver should wait before triggering an RX_TIMEOUT event, when set into RX_SINGLE mode.

These commits were cherry-picked from #11022 



### Testing procedure
Compile and flash the `drivers_sx127x` test:
```
BOARD=b-l072z-lrwan1 make -C tests/driver_sx127x all flash term
```

Then:
- Verify the `random` command give random numbers
- Use the `syncword` command to get and set the syncword.
    - Set a syncword (in decimal) with "syncword set <dec>"
    - Get the syncword (in hex) with "syncword get". Check both numbers are equivalent
    - Also, check the same hex number in 0x39 register with `register get 0x39`
- Use the `rx_timeout` command to set the RX symbol timeout
    - `rx_timeout set <dec>`
    - Read the content of the registers 0x1E and 0x1F (rx_timeout is 10 bits long and scattered in these 2 registers). The 2 MSB of the rx_timeout are the 2 LSB of the 0x1E register and the 8 LSB are in the 0x1F reg. Check the hex matches the introduced rx_timeout value.
E.g
```1561389398.684083;st-lrwan1-11;> rx_timeout set 50
1561389398.686839;st-lrwan1-11;New rx_timeout set
st-lrwan1-11;register get 0x1E
1561389453.166605;st-lrwan1-11;> register get 0x1E
1561389453.168076;st-lrwan1-11;[regs] 0x1E = 0xC4
st-lrwan1-11;register get 0x1F
1561389457.242929;st-lrwan1-11;> register get 0x1F
1561389457.245584;st-lrwan1-11;[regs] 0x1F = 0x32
```
note ((0xC4 & 0x3) << 8) | 0x32 == 0x32 => 50 decimal




### Issues/PRs references
